opentracing/tracecontext.go

```diff
@@ -51,18 +43,18 @@ type TraceContext interface {
 	// `restrictedKey` MUST match the regular expression
 	// `(?i:[a-z0-9][-a-z0-9]*)` and is case-insensitive. That is, it must
 	// start with a letter or number, and the remaining characters must be
-	// letters, numbers, or hyphens. See CanonicalizeTraceTagKey(). If
-	// `restrictedKey` does not meet these criteria, SetTraceTag() results in
-	// undefined behavior.
+	// letters, numbers, or hyphens. See CanonicalizeTraceAttributeKey(). If
+	// `restrictedKey` does not meet these criteria, SetTraceAttribute()
+	// results in undefined behavior.
 	//
 	// Returns a reference to this TraceContext for chaining, etc.
-	SetTraceTag(restrictedKey, value string) TraceContext
+	SetTraceAttribute(restrictedKey, value string) TraceContext
```

Comment:
I still think upstream propagation semantic is fairly intuitive. It has certain limitations that people must understand before using it, of course.

> there's no formal requirement (in the data model) that children finish before their parents.

If an RPC is made with expectation of a result returned to the caller, then I think it's reasonable to assume the parent span must last longer than the children. Caveat emptor.

@bensigelman are you suggesting PropagateAttributeDown as a method name?  Why not this:
```
setTraceAttribute(key, value, propagationMode)
``